quic_transport/sdk/impl/quic_transport_owt_server_impl.cc

```diff
@@ -0,0 +1,288 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "owt/quic_transport/sdk/impl/quic_transport_owt_server_impl.h"
+
+#include <string.h>
+
+#include "base/location.h"
+#include "base/threading/thread_task_runner_handle.h"
+#include "net/base/ip_endpoint.h"
+#include "net/base/net_errors.h"
+#include "net/log/net_log_source.h"
+#include "net/socket/udp_server_socket.h"
+#include "net/third_party/quiche/src/quiche/quic/core/crypto/crypto_handshake.h"
+#include "net/third_party/quiche/src/quiche/quic/core/crypto/quic_random.h"
+#include "net/third_party/quiche/src/quiche/quic/core/quic_crypto_stream.h"
+#include "net/third_party/quiche/src/quiche/quic/core/quic_data_reader.h"
+#include "net/third_party/quiche/src/quiche/quic/core/quic_packets.h"
+#include "net/third_party/quiche/src/quiche/quic/core/tls_server_handshaker.h"
+#include "net/tools/quic/quic_simple_server_packet_writer.h"
+#include "net/tools/quic/quic_simple_server_session_helper.h"
+#include "net/quic/address_utils.h"
+
+namespace net {
+
+namespace {
+
+const char kSourceAddressTokenSecret[] = "secret";
```

Comment:
what's this

Reply:
secret key material used for encrypting and decrypting source address tokens. It can be of any length as it is fed into a KDF before use.  used by constructor defined in net/third_party/quiche/src/quiche/quic/core/crypto/quic_crypto_server_config.h

Reply:
hard coded secret ?

Reply:
Yes, it's a hard coded key